File: client/rating-app-client/src/app/rated-game.ts
import { WithId } from './with-id';
import { User } from './user';
import { Rating } from './rating';

export class RatedGame {
  winner: WithId<User>;
  winner_rating: WithId<Rating>;
  loser: WithId<User>;
  loser_rating: WithId<Rating>;
  date_played: Date;
}

File: server/src/main/scala/rating/models/RatedGame.scala
package rating.models

import rating.repositories.WithId
import io.circe.generic.JsonCodec, io.circe.syntax._

@JsonCodec case class RatedGame(
  winner: WithId[User],
  winner_rating: WithId[Rating],
  loser: WithId[User],
  loser_rating: WithId[Rating],
  date_played: Timestamp
)
